Automattic/jetpack
# Eventbrite Block: Update saved block markup

Fixes #14506

#### Changes proposed in this Pull Request:

* When the Eventbrite uses the Button & Modal embed type, add `.wp-block-button` to the block wrapper in order to appropriately style the `.wp-block-button__link` inside.
* Change the markup to be `<a role="button">` rather than `<button>` to be compatible with WordPress.com `wp_kses` configuration.

| Before | After |
| - | - |
| <img width="629" alt="Screen Shot 2020-01-28 at 19 54 56" src="https://user-images.githubusercontent.com/1699996/73321453-194efd00-4208-11ea-847f-ef3cd4367a0e.png"> | <img width="555" alt="Screenshot 2020-01-29 at 13 50 01" src="https://user-images.githubusercontent.com/2070010/73362588-01768800-429f-11ea-8cc2-61b13d003cb8.png"> |

⚠️ Note: this is change breaks existing blocks. Since the block is not released yet, I haven't created a deprecation/migration path. Just keep in mind that you might need to let Gutenberg try to automatically fix any existing blocks, or just remove them and add new ones.

**Additional Context**

In Core Guten, `wp-block-button__link` is the inner element inside a `wp-block-button`.
So, it's to be expected that it malfunctions when used "outside" `wp-block-button`.

E.g. this is how a normal Button block is rendered in the editor:
<img width="921" alt="Screenshot 2020-01-29 at 13 31 50" src="https://user-images.githubusercontent.com/2070010/73360952-e0606800-429b-11ea-83dd-ee617fde9fd5.png">

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* Bug fix

#### Testing instructions:






* Open the Editor, insert an Eventbrite block, and provide an Eventbrite event URL.
* From the block sidebar, change the Embed Type to "Button & Modal".
* Make sure the previewed button style is consistent to the current theme, instead of being dark grey with black text color.
* Save and view the post.
* Make sure the rendered button is identical to how it looks in the editor.
* Modify the button style from the block sidebar (make it outline, change the colors and the border radius).
* Make sure the button is updated accordingly in both the editor and the front end.

#### Proposed changelog entry for your changes:

* N/A
